keras/callbacks.py

```diff
@@ -792,12 +797,31 @@ def on_epoch_end(self, epoch, logs=None):
             summary_value = summary.value.add()
             summary_value.simple_value = value.item()
             summary_value.tag = name
-            self.writer.add_summary(summary, epoch)
+            self.writer.add_summary(summary, self.seen)
         self.writer.flush()
+        self.seen += self.batch_size
 
     def on_train_end(self, _):
         self.writer.close()
 
+    def on_batch_end(self, batch, logs=None):
+        logs = logs or {}
+
+        if self.write_batch_performance:
+            for name, value in logs.items():
+                if name in ['batch', 'size']:
+                    continue
+                summary = tf.Summary()
+                summary_value = summary.value.add()
+                summary_value.simple_value = value.item()
+                summary_value.tag = name
+                self.writer.add_summary(summary, self.seen)
+            self.writer.flush()
+
+        self.seen += self.batch_size
+    # @todo write some unit tests for this
```

Comment:
I believe you can remove the todo here, since you set it to true in the unit test